client/landing/jetpack-cloud/sections/settings/main.jsx

```diff
@@ -27,55 +29,95 @@ import './style.scss';
+		args: [ disconnectedMessage ],
+		components: {
+			a: (
+				<ExternalLink
+					className="settings__link-external"
+					icon
+					href="https://jetpack.com/support/adding-credentials-to-jetpack/"
+				/>
+			),
+		},
+	} ),
 } );
 
-const ConnectionStatus = ( { isConnected } ) => {
-	const translate = useTranslate();
-	const cardProps = isConnected ? connectedProps( translate ) : disconnectedProps( translate );
-
-	return (
-		<Card compact={ true } className={ cardProps.className }>
-			<img className="settings__icon" src={ cardProps.iconPath } alt="" />
-			<div className="settings__details">
-				<div className="settings__details-head"> { cardProps.title } </div>
-				<div>{ cardProps.content }</div>
-			</div>
-		</Card>
-	);
+const getCardProps = ( isConnected, message, translate ) => {
+	if ( isConnected ) {
+		return connectedProps( translate, message );
+	}
+	return disconnectedProps( translate, message );
 };
 
+const ConnectionStatus = ( { cardProps } ) => (
+	<Card compact={ true } className={ cardProps.className }>
+		<img className="settings__icon" src={ cardProps.iconPath } alt="" />
+		<div className="settings__details">
+			<div className="settings__details-head"> { cardProps.title } </div>
+			<div>{ cardProps.content }</div>
+		</div>
+	</Card>
+);
+
 const SettingsPage = () => {
 	const translate = useTranslate();
 	const siteId = useSelector( getSelectedSiteId );
 
-	const rewind = useSelector( ( state ) => getRewindState( state, siteId ) );
-	const isInitialized = rewind.state !== 'uninitialized';
-	const isConnected = rewind.state === 'active';
+	const NO_PRODUCT = 0;
+	const HAS_BACKUP = 1;
+	const HAS_SCAN = 2;
+	const HAVE_BOTH = 3;
+
+	const DISCONNECTED = 0;
+	const CONNECTED = 1;
+
+	const messages = {
+		[ DISCONNECTED ]: {
+			[ NO_PRODUCT ]: '',
+			[ HAS_BACKUP ]: translate(
+				'Enter your server credentials to enable one-click restores for backups.'
+			),
+			[ HAS_SCAN ]: translate(
+				'Enter your server credentials to enable Jetpack to auto fix threats.'
+			),
+			[ HAVE_BOTH ]: translate(
+				'Enter your server credentials to enable one-click restores for backups and to auto-fix threats.'
+			),
+		},
+		[ CONNECTED ]: {
+			[ NO_PRODUCT ]: '',
+			[ HAS_BACKUP ]: translate( 'One-click restores are enabled.' ),
+			[ HAS_SCAN ]: translate( 'Auto-fix threats are enabled.' ),
+			[ HAVE_BOTH ]: translate( 'One-click restores and auto-fix threats are enabled.' ),
+		},
+	};
+
+	const scanState = useSelector( ( state ) => getSiteScanState( state, siteId ) );
+	const backupState = useSelector( ( state ) => getRewindState( state, siteId ) );
+
+	const isInitialized = backupState.state !== 'uninitialized';
+	const isConnected = backupState.state === 'active';
+
+	const hasBackup = backupState?.state !== 'unavailable';
+	const hasScan = scanState?.state !== 'unavailable';
+
+	const productCode = hasBackup | ( hasScan << 1 );
+
+	const message = messages[ +isConnected ][ productCode ];
```

Comment:
const productCode = hasBackup | ( hasScan << 1 );
is a bit hard to understand what productCode is. 

+isConnected is a bit confusing for me. 

Can we gave a function that simplifies this a bit?

Reply:
Yes, I thought to add some comments to it, I forgot. It's a binary operation, to have this 01 for backup only, 10 to scan only, and 11 for both.

isConnected is a boolean, so to convert it to a number we can use that operator.

Reply:
@enejb updated